ghost/adapter-cache-redis/lib/AdapterCacheRedis.js

```diff
@@ -166,6 +186,26 @@ class AdapterCacheRedis extends BaseCacheAdapter {
      * @param {() => Promise<any>} [fetchData] An optional function to fetch the data, which will be used in the case of a cache MISS or a background refresh
      */
     async get(key, fetchData) {
+        const internalKey = this._buildKey(key);
+        if (this.currentlyExecutingReads.has(internalKey)) {
```

Comment:
I think there's possibly a change of a race condition here? Where `.has` passes, but the Promise is deleted before we get to the `.get`? Maybe?

Would this work better?

```
const currentlyExecutingPromise = this.currentlyExecutingReads.get(internalKey);
if (currentlyExecutingPromise) {
	return currentlyExecutingPromise.promise;
}
```